File: MovingAverageInterface.py
class MovingAverageInterface:
    """Data Structure Interface for Moving Average"""

    def __init__(self):
        """Initialize Moving Average data structure"""
        pass

    def getElements(self):
        """Get access to the elements of structure"""
        pass

    def addElements(self, element):
        """Add element to the structure.
        :param element
        """
        pass

    def calculateMovingAverage(self, lastN):
        """Moving average of the last 'N' elements added to the structure
        :type lastN: object
        """
        pass

File: MovingAverageInterface.java
public interface MovingAverageInterface {

    /**
     * add element to the data structure
     * @param element
     */
    public void addElement(Double element);

    /**
     * returns element present at the specified position
     * @param position
     * @return element
     */
    public Double getElement (int position);

    /**
     * calculates the average of the last N elements
     * @return average
     */
    public Double getAverage();
}
